### ISSUE TITLE
DateTime Invalid values

### ISSUE THREAD

**Issue by samuelGrahame** [bridgedotnet/Bridge#2261]
A description of the issue. 

### Expected

```js
0001-01-01
0001-01-01
0001-01-01
0001-01-01
```

### Actual

```js
1901-01-01
1901-01-01
-271821-04-20
2001-01-01
```

### Steps To Reproduce

http://deck.net/4fde911d3bdc7a131a9cffbaa0c4d6ab

```c#
public class Program
{
    public static void Main()
    {
        PrintDate(DateTime.Parse("0001-01-01"));
        PrintDate(new DateTime(1, 1, 1));
        PrintDate(DateTime.MinValue);
        
        PrintDate(DateTime.Parse("0001-01-01 00:00:00"));
    }

    public static void PrintDate(DateTime dateTime)
    {
        Console.WriteLine(dateTime.ToString("yyyy-MM-dd"));
    }

}
```

**Comment by LeonidVeriga** [comment]
Related to #1606

**Comment by geoffreymcgill** [comment]
Fixed in #1606.
